From e640a086b1af37ef77519ef4fc6b50d6b26b6741 Mon Sep 17 00:00:00 2001
From: b07117 <b07117@freescale.com>
Date: Fri, 9 Sep 2011 10:51:06 +0800
Subject: [PATCH 175/486] ENGR00156308, Re-enable video switch between primary
 and second displays

video renderer introduce hw composer, need re-enable it

Signed-off-by: xianzhong <b07117@freescale.com>
---
 mx5x/hwcomposer/hwcomposer.cpp | 21 +++++++++++++++++++++
 mx6/hwcomposer/hwcomposer.cpp  | 21 +++++++++++++++++++++
 2 files changed, 42 insertions(+)

diff --git a/mx5x/hwcomposer/hwcomposer.cpp b/mx5x/hwcomposer/hwcomposer.cpp
index 7ff9e78..38782df 100755
--- a/mx5x/hwcomposer/hwcomposer.cpp
+++ b/mx5x/hwcomposer/hwcomposer.cpp
@@ -85,6 +85,27 @@ static int hwc_check_property(hwc_context_t *dev)
     //bool bValue = false;
     char value[10];
     int orignMode = dev->display_mode;
+
+    property_get("rw.VIDEO_TVOUT_DISPLAY", value, "");
+    if (strcmp(value, "1") == 0) {
+        property_set("rw.VIDEO_OVERLAY_DISPLAY", "0");
+        property_set("sys.VIDEO_DISPLAY", "1");
+    }
+    else
+    {
+       property_set("rw.VIDEO_OVERLAY_DISPLAY", "1");
+       property_set("sys.VIDEO_DISPLAY", "0");
+    }
+    property_get("sys.SECOND_DISPLAY_ENABLED", value, "");
+    if (strcmp(value, "1") == 0) {
+       property_set("rw.VIDEO_OVERLAY_DISPLAY", "2");
+       property_set("sys.VIDEO_DISPLAY", "0");
+    } else
+    {
+       property_set("rw.VIDEO_OVERLAY_DISPLAY", "1");
+       property_set("sys.VIDEO_DISPLAY", "0");
+    }
+
     /*note:rw.VIDEO_OVERLAY_DISPLAY means the overlay will be combined to which display.
      *the default value is 0 and it indicates nothing.
      *if the value is 1 and it indicates combined to display0.
diff --git a/mx6/hwcomposer/hwcomposer.cpp b/mx6/hwcomposer/hwcomposer.cpp
index 147a818..cdce636 100755
--- a/mx6/hwcomposer/hwcomposer.cpp
+++ b/mx6/hwcomposer/hwcomposer.cpp
@@ -85,6 +85,27 @@ static int hwc_check_property(hwc_context_t *dev)
     //bool bValue = false;
     char value[10];
     int orignMode = dev->display_mode;
+
+    property_get("rw.VIDEO_TVOUT_DISPLAY", value, "");
+    if (strcmp(value, "1") == 0) {
+        property_set("rw.VIDEO_OVERLAY_DISPLAY", "0");
+        property_set("sys.VIDEO_DISPLAY", "1");
+    }
+    else
+    {
+       property_set("rw.VIDEO_OVERLAY_DISPLAY", "1");
+       property_set("sys.VIDEO_DISPLAY", "0");
+    }
+    property_get("sys.SECOND_DISPLAY_ENABLED", value, "");
+    if (strcmp(value, "1") == 0) {
+       property_set("rw.VIDEO_OVERLAY_DISPLAY", "2");
+       property_set("sys.VIDEO_DISPLAY", "0");
+    } else
+	{
+       property_set("rw.VIDEO_OVERLAY_DISPLAY", "1");
+       property_set("sys.VIDEO_DISPLAY", "0");
+	}
+
     /*note:rw.VIDEO_OVERLAY_DISPLAY means the overlay will be combined to which display.
      *the default value is 0 and it indicates nothing.
      *if the value is 1 and it indicates combined to display0.
-- 
1.8.0

